reco_utils/dataset/wikidata.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import requests
+
+def find_wikidataID(name):
+    """Find the entity ID in wikidata from a title string.
+    Args:
+        name (str): A string with a wikidata page title.
+    Returns:
+        entityID: wikidata entityID corresponding to the title string. 
+                  'entityNotFound' will be returned if no page is found
+    """
+    url = "https://en.wikipedia.org/w/api.php?action=query&prop=pageprops&format=json&titles="
+    r = requests.get(url+name.replace(" ", "%20"))
+    entityID = list(r.json()["query"]["pages"].keys())[0]
+    try:
+        entityID = r.json()["query"]["pages"][entityID]["pageprops"]["wikibase_item"]
+    except:
+        entityID = "entityNotFound"
+    return entityID
+
+def query_entity_links(entityID):
+    """Query all linked pages from a wikidata entityID
+    Args:
+        entityID (str): A wikidata page ID.
+    Returns:
+        data (json): dictionary with linked pages.
+    """
+    query = """
+    PREFIX entity: <http://www.wikidata.org/entity/>
+    #partial results
+
+    SELECT ?propUrl ?propLabel ?valUrl ?valLabel
+    WHERE
+    {
+        hint:Query hint:optimizer 'None' .
+        {	BIND(entity:"""+entityID+""" AS ?valUrl) .
+            BIND("N/A" AS ?propUrl ) .
+            BIND("identity"@en AS ?propLabel ) .
+        }
+        UNION
+        {	entity:"""+entityID+""" ?propUrl ?valUrl .
+            ?property ?ref ?propUrl .
+            ?property rdf:type wikibase:Property .
+            ?property rdfs:label ?propLabel
+        }
+
+        ?valUrl rdfs:label ?valLabel
+        FILTER (LANG(?valLabel) = 'en') .
+        OPTIONAL{ ?valUrl wdt:P18 ?picture .}
+        FILTER (lang(?propLabel) = 'en' )
+    }
+    ORDER BY ?propUrl ?valUrl
+    LIMIT 500
+    """
+    url = 'https://query.wikidata.org/sparql'
+    r = requests.get(url, params = {'format': 'json', 'query': query})
```

Comment:
is there a query limit for this? if so, we might want to use lru_cache

Reply:
In the Query Limits section (https://www.mediawiki.org/wiki/Wikidata_Query_Service/User_Manual) of the documentation they mention: "the service is limited to 5 parallel queries per IP". Seems like the only limit is in parallel queries per IP, not in sequential queries